jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/StatisticsHandler.java

```diff
@@ -58,16 +53,9 @@ public boolean process(Request request, Response response, Callback callback) th
         if (next == null)
             return false;
 
-        StatisticsRequest statisticsRequest = newStatisticsRequest(request);
+        Request statisticsRequest = newStatisticsRequest(request);
         try
         {
-            _processStats.increment();
-            _requestStats.increment();
-
-            String id = statisticsRequest.getConnectionMetaData().getId();
-            if (_connectionStats.add(id))
-                statisticsRequest.getConnectionMetaData().getConnection().addEventListener(statisticsRequest);
-
             if (next.process(statisticsRequest, response, callback))
                 return true;
             _requestStats.decrement();
```

Comment:
I see your point, and it's fair: `StatisticsRequest` should have `protected` visibility. But since decrementing this counter cannot be done by the `StatisticsRequest` itself as only the handler's `process` method has this knowledge, I've added a `StatisticsRequest.unProcessed()` method  that does this decrement and documented that it must be called in case the wrapped handler does not process the request.